api/engine.py

```diff
@@ -318,6 +324,73 @@ def __get_result_content():
                                    msg="your scan id is not valid!")), 400
     return __get_result(__language(), id)
 
+@app.route("/results/get_json", methods=["GET"])
+def __get_results_json():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        id = __get_value(flask_request, "id")
+    except:
+        id = ""
+    result_id = session.query(Report).join(HostsLog, Report.id==HostsLog.id).filter(Report.id==id).all()
```

Comment:
join here is wrong hostlog.id is not related to report.id

Reply:
@aman566 you cannot join report.id and hostslog.id - they are two different identities/primary keys. still getting wrong data. Imagine you have empty reports table and empty hosts table (just installed Nettacker). now imagine you run a port_scan for port 80 on all subdomains of uber.com. after the scan reports table will have 1 record with id=1 and hostslog table will have 326 records with id field having values  from 1 to 325 (326 rows in table hostslog, but only one record in reports). If you join on id you will get just the first host (subdomain of uber.com) in the report. You should join on scan_id.